packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -311,6 +311,11 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate {
 
   updateItemSize(key: Key, size: Size) {
     let layoutInfo = this.layoutInfos.get(key);
+    // If no layoutInfo, item has been deleted/removed.
+    if (!layoutInfo) {
+      return false;
+    }
```

Comment:
Alternative/supplemental fix: maybe change https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/virtualizer/src/Virtualizer.ts#L907 to something like
```
let changed = !this.layout.getLayoutInfo(key) || this.layout.updateItemSize(key, size);
```
aka if that key doesn't have layout info, don't bother calling updateItemSize and perform a relayout (or maybe don't relayout?)